### PR TITLE
[EJBCLIENT-162] EJB client unnecessary mark a channel/connection as broken if can't read a message

### PR DESCRIPTION
**Pull request by gaol**
Jira: https://issues.jboss.org/browse/EJBCLIENT-162
EAP 7.0 Jira: https://issues.jboss.org/browse/JBEAP-5411
EAP 7.1 Jira: https://issues.jboss.org/browse/JBEAP-5413 - Verified 

This issue does not happen anymore in EAP 7.1 after updating jboss-ejb-client to 4.x, the related errors are removed during the refracting in jboss-ejb-client master branch(4.x).

So there is no corresponding PR for this issue in master branch.

### CODE REVIEW

File: src/main/java/org/jboss/ejb/client/EJBInvocationHandler.java [hunk 1]
```diff
@@ -277,12 +279,11 @@ private static void sendRequestWithPossibleRetries(final EJBClientInvocationCont
             }
             // retry the request
             final String failedNodeName = rsfe.getFailedNodeName();
-            if (failedNodeName != null) {
+            if (failedNodeName != null && !failedNodes.contains(failedNodeName)) {
+                failedNodes.add(failedNodeName);
                 Logs.MAIN.debugf(rsfe, "Retrying invocation %s which failed on node: %s due to:", clientInvocationContext, failedNodeName);
-                // exclude this failed node, during the retry
-                clientInvocationContext.markNodeAsExcluded(failedNodeName);
```
Comment (gaol) [inline src/main/java/org/jboss/ejb/client/EJBInvocationHandler.java]: What customer complained was that "Further EJB invocations are rejected with EJBCLIENT0025". Don't exclude the node in the retry process will prevent that.

**Comment by gaol** [comment]
@dmlloyd Could you please take a look at this PR? Thank you very much!

**Comment by dmlloyd** [comment]
I think we need an upstream (2.x, 3.x) PR for this as well.

**Comment by gaol** [comment]
Thank you to taking care of this PR.
* PR for `2.x` branch: https://github.com/wildfly/jboss-ejb-client/pull/296
* PR for `3.x` branch: https://github.com/wildfly/jboss-ejb-client/pull/295

**Pull request merged**